pkg/block/block.go

```diff
@@ -145,11 +145,11 @@ func upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir st
 		return errors.Wrap(err, "encode meta file")
 	}
 
-	if err := objstore.UploadDir(ctx, logger, bkt, path.Join(bdir, ChunksDirname), path.Join(id.String(), ChunksDirname)); err != nil {
+	if err := objstore.UploadDir(ctx, logger, bkt, filepath.Join(bdir, ChunksDirname), path.Join(id.String(), ChunksDirname)); err != nil {
```

Comment:
`path.Join()` is correct in this file because on object storages it's always `/`. If I remember correctly, the objstore client API contract is to always use `/`.

_Same comment applies below._

Reply:
@pracucci but the `srcdir` argument to `objstore.UploadDir` refers to a filesystem directory, not an object storage one?

Reply:
Oh sorry, you're definitely right!